timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/LocalPaxosComponents.java

```diff
@@ -100,6 +102,31 @@ public static LocalPaxosComponents createWithBlockingMigration(
             boolean canCreateNewClients,
             OrderableSlsVersion timeLockVersion,
             boolean skipConsistencyCheckAndTruncateOldPaxosLog) {
+        ExecutorService sqliteAsyncExecutor = PTExecutors.newSingleThreadExecutor(true);
+        LocalPaxosComponents components = createWithAsyncMigration(
+                metrics,
+                paxosUseCase,
+                legacyLogDirectory,
+                sqliteDataSource,
+                leaderUuid,
+                canCreateNewClients,
+                timeLockVersion,
+                skipConsistencyCheckAndTruncateOldPaxosLog,
+                sqliteAsyncExecutor);
+        sqliteAsyncExecutor.shutdown();
```

Comment:
can we do this in a finally block? otherwise if `createWithAsyncMigration` throws this may not run